src/main/kotlin/org/sonar/fakeplugin/language/SimpleAnnotator.kt

```diff
@@ -0,0 +1,41 @@
+package org.sonar.fakeplugin.language
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.editor.DefaultLanguageHighlighterColors
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.elementType
+
+class SimpleAnnotator : Annotator {
+
+    companion object {
+        const val WORD_ANNOTATED = "SONAR"
+    }
+
+    override fun annotate(element: PsiElement, holder: AnnotationHolder) {
+        val value = element.text ?: return
+        val textRange = element.textRange
+        val elementType = element.elementType.toString()
+
+        if (elementType == "PACKAGE_DIRECTIVE") {
```

Comment:
Have you found this kind of pattern in official documentation? I would be tempted to prefer relying on Java types than on magical constants. Something like:
`if (element is PsiPackage)`

Reply:
Agree it's terrible practice - I couldn't find proper documentation for this so I wanted feedback)

Might be wrong - but since I'm using Kotlin language it doesn't resolve to standard PSI classes, but rather Kotlin PSI classes like `KtDirectivePackage` for example, which I don't have in my class loader

I'm trying to make use of Kotlin plugin and dependencies, but still WIP

Reply:
Fixed by adding Kotlin plugin in Intellij + Kotlin dependency in plugin.xml